plugins/ltac/rewrite.ml

```diff
@@ -1898,7 +1899,7 @@ let build_morphism_signature env sigma m =
   let m,ctx = Constrintern.interp_constr env sigma m in
   let m = EConstr.of_constr m in
   let sigma = Evd.from_ctx ctx in
-  let t = Typing.unsafe_type_of env sigma m in
+  let sigma, t = Typing.type_of env sigma m in
```

Comment:
We just interpreted m, no need for re-typing it, actually here I think the interp_constr interface is somewhat wrong, it should just return the inferred type instead of dropping it